### ISSUE TITLE
Bug: Debezium JSON parser cannot parse row with upper case column name

### ISSUE THREAD

**Issue by StrikeW** [risingwavelabs/risingwave#10481]
### Describe the bug

MySQL cdc integration test failed due to parsing errors and rows have been skipped. The test case can pass on last [Thursday](https://buildkite.com/risingwavelabs/integration-tests/builds/142), but fails after PR #10096 has been merged.

### Error message/log

```text
2023-06-21T19:51:07.908144+08:00 DEBUG risingwave_connector::parser::unified::util: inserted "l_orderkey" Int64 Err(Undefined { name: "l_orderkey", path: "after" })
2023-06-21T19:51:08.191859+08:00  WARN risingwave_connector::parser: failed to parse source data: internal error: AccessError: Undefined { name: "l_orderkey", path: "after" }
2023-06-21T19:51:08.191931+08:00  WARN risingwave_connector::parser: message parsing failed internal error: AccessError: Undefined { name: "l_orderkey", path: "after" }, skipping
2023-06-21T19:51:08.192052+08:00 DEBUG risingwave_connector::parser::unified::util: inserted "l_orderkey" Int64 Err(Undefined { name: "l_orderkey", path: "after" }
```

### To Reproduce

Start mysql containers and generate test data:
```bash
docker compose -f integration_tests/mysql-cdc/docker-compose.yml up -d mysql
docker compose -f integration_tests/mysql-cdc/docker-compose.yml up -d datagen_tpch
```

Launch the risingwave cluster and run the sql:
```sql
CREATE TABLE lineitem_rw (
   L_ORDERKEY BIGINT,
   L_PARTKEY BIGINT,
   L_SUPPKEY BIGINT,
   L_LINENUMBER BIGINT,
   L_QUANTITY DECIMAL,
   L_EXTENDEDPRICE DECIMAL,
   L_DISCOUNT DECIMAL,
   L_TAX DECIMAL,
   L_RETURNFLAG VARCHAR,
   L_LINESTATUS VARCHAR,
   L_SHIPDATE DATE,
   L_COMMITDATE DATE,
   L_RECEIPTDATE DATE,
   L_SHIPINSTRUCT VARCHAR,
   L_SHIPMODE VARCHAR,
   L_COMMENT VARCHAR,
   PRIMARY KEY(L_ORDERKEY, L_LINENUMBER)
) WITH (
    connector = 'mysql-cdc',
    hostname = '127.0.0.1',
    port = '8306',
    username = 'root',
    password = '123456',
    database.name = 'mydb',
    table.name = 'lineitem',
    server.id = '2'
);
```

### How did you deploy RisingWave?

`./risedev d`

### Additional context

Debezium JSON for some upstream rows for reference.
```
2023-06-21 19:50:58,047 INFO  [rw-dbz-engine-runner-10002] core.DbzCdcEventConsumer:107 - record => {"before":null,"after":{"L_ORDERKEY":56165,"L_PARTKEY":39492,"L_SUPPKEY":4499,"L_LINENUMBER":1,"L_QUANTITY":39.0,"L_EXTENDEDPRICE":55828.11,"L_DISCOUNT":0.0,"L_TAX":0.04,"L_RETURNFLAG":"R","L_LINESTATUS":"F","L_SHIPDATE":"1992-06-22","L_COMMITDATE":"1992-06-11","L_RECEIPTDATE":"1992-06-28","L_SHIPINSTRUCT":"NONE","L_SHIPMODE":"RAIL","L_COMMENT":"s. packages sleep s"},"source":{"version":"1.9.7.Final","connector":"mysql","name":"RW_CDC_mydb.lineitem.10002","ts_ms":1687348255000,"snapshot":"true","db":"mydb","sequence":null,"table":"lineitem","server_id":0,"gtid":null,"file":"binlog.000003","pos":7424302,"row":0,"thread":null,"query":null},"op":"r","ts_ms":1687348257797,"transaction":null}
2023-06-21 19:50:58,047 INFO  [rw-dbz-engine-runner-10002] core.DbzCdcEventConsumer:107 - record => {"before":null,"after":{"L_ORDERKEY":56166,"L_PARTKEY":148713,"L_SUPPKEY":1228,"L_LINENUMBER":1,"L_QUANTITY":11.0,"L_EXTENDEDPRICE":19378.81,"L_DISCOUNT":0.04,"L_TAX":0.02,"L_RETURNFLAG":"A","L_LINESTATUS":"F","L_SHIPDATE":"1993-11-07","L_COMMITDATE":"1993-10-27","L_RECEIPTDATE":"1993-12-04","L_SHIPINSTRUCT":"TAKE BACK RETURN","L_SHIPMODE":"FOB","L_COMMENT":"cial requests cajole slyly regular tithes."},"source":{"version":"1.9.7.Final","connector":"mysql","name":"RW_CDC_mydb.lineitem.10002","ts_ms":1687348255000,"snapshot":"true","db":"mydb","sequence":null,"table":"lineitem","server_id":0,"gtid":null,"file":"binlog.000003","pos":7424302,"row":0,"thread":null,"query":null},"op":"r","ts_ms":1687348257797,"transaction":null}
2023-06-21 19:50:58,047 INFO  [rw-dbz-engine-runner-10002] core.DbzCdcEventConsumer:107 - record => {"before":null,"after":{"L_ORDERKEY":56166,"L_PARTKEY":10056,"L_SUPPKEY":5059,"L_LINENUMBER":2,"L_QUANTITY":5.0,"L_EXTENDEDPRICE":4830.25,"L_DISCOUNT":0.1,"L_TAX":0.06,"L_RETURNFLAG":"R","L_LINESTATUS":"F","L_SHIPDATE":"1993-09-15","L_COMMITDATE":"1993-10-14","L_RECEIPTDATE":"1993-09-30","L_SHIPINSTRUCT":"COLLECT COD","L_SHIPMODE":"REG AIR","L_COMMENT":"ly ironic theodol"},"source":{"version":"1.9.7.Final","connector":"mysql","name":"RW_CDC_mydb.lineitem.10002","ts_ms":1687348255000,"snapshot":"true","db":"mydb","sequence":null,"table":"lineitem","server_id":0,"gtid":null,"file":"binlog.000003","pos":7424302,"row":0,"thread":null,"query":null},"op":"r","ts_ms":1687348257797,"transaction":null}
```

**Comment by StrikeW** [comment]
cc @tabVersion @idx0-dev

**Comment by idx0-dev** [comment]
I have made a fix in #10482. Please help me to triggler the integrations on that branch to verify. ❤️ @StrikeW

**Comment by StrikeW** [comment]
Fixed by https://github.com/risingwavelabs/risingwave/pull/10482
